pkg/agent/agent.go

```diff
@@ -443,6 +445,10 @@ func (a *agent) ManageOwnTaints(ctx context.Context, resourceState []DrbdResourc
 	if len(eventUpdates) != 0 {
 		klog.V(1).Info("updating node taints")
 
+		if a.Options.DisableNodeTaints {
+			updatedTaints = []corev1.Taint{}
+		}
+
```

Comment:
This effectively _removes all_ taints from a node, not just the ones added by the HA Controller.

Would it not be much simpler to have this check at the top of the function, `return nil` and maybe log some message at debug level that taints are not updated because of the settings?